tui/popup_passphrase.go

```diff
@@ -0,0 +1,239 @@
+	dialog.infoLabel.SetMultiline(true)
+
+	dialog.passphraseEdit = clui.CreateEditField(borderFrame, 1, "", Fixed)
+	dialog.passphraseEdit.SetPasswordMode(true)
+
+	dialog.passphraseEdit.OnChange(func(ev clui.Event) {
+		dialog.validatePassphrase()
+	})
+
+	dialog.passphraseEdit.OnActive(func(active bool) {
+		if dialog.passphraseEdit.Active() {
+			dialog.validatePassphrase()
+		}
+	})
+
+	dialog.passphraseEdit.OnKeyPress(func(k term.Key, ch rune) bool {
+		if k == term.KeyCtrlU {
+			dialog.revealPassphrase()
+			return true
+		}
+		if k == term.KeyArrowUp || k == term.KeyArrowDown {
+			return false
+		}
+		if !dialog.changedPassphrase {
+			dialog.changedPassphrase = true
+			dialog.passphraseEdit.SetTitle("")
+			dialog.ppConfirmEdit.SetTitle("")
+		}
+		return false
+	})
+
+	dialog.ppConfirmEdit = clui.CreateEditField(borderFrame, 1, "", Fixed)
+	dialog.ppConfirmEdit.SetPasswordMode(true)
+	dialog.warningLabel = clui.CreateLabel(borderFrame, AutoSize, 1, "", Fixed)
+	dialog.warningLabel.SetMultiline(true)
+	dialog.warningLabel.SetBackColor(errorLabelBg)
+	dialog.warningLabel.SetTextColor(errorLabelFg)
+
+	dialog.ppConfirmEdit.OnChange(func(ev clui.Event) {
+		dialog.validatePassphrase()
+	})
+
+	dialog.ppConfirmEdit.OnActive(func(active bool) {
+		if dialog.ppConfirmEdit.Active() {
+			dialog.validatePassphrase()
+		}
+	})
+
+	dialog.ppConfirmEdit.OnKeyPress(func(k term.Key, ch rune) bool {
+		if k == term.KeyCtrlU {
+			dialog.revealPassphrase()
+			return true
+		}
+		if k == term.KeyArrowUp || k == term.KeyArrowDown {
+			return false
+		}
+		if !dialog.changedPassphrase {
+			dialog.changedPassphrase = true
+			dialog.passphraseEdit.SetTitle("")
+			dialog.ppConfirmEdit.SetTitle("")
+		}
+		return false
+	})
+
+	buttonFrame := clui.CreateFrame(borderFrame, AutoSize, 1, clui.BorderNone, clui.Fixed)
+	buttonFrame.SetPack(clui.Horizontal)
+	buttonFrame.SetGaps(1, 0)
+	dialog.cancelButton = CreateSimpleButton(buttonFrame, AutoSize, AutoSize, "Cancel", Fixed)
+	dialog.cancelButton.SetEnabled(true)
+
+	dialog.confirmButton = CreateSimpleButton(buttonFrame, AutoSize, AutoSize, "Confirm Install", Fixed)
+
+	return nil
+}
+
+// CreateEncryptPassphraseDialogBox creates the Network PopUp
+func CreateEncryptPassphraseDialogBox(modelSI *model.SystemInstall) (*EncryptPassphraseDialog, error) {
+	dialog := new(EncryptPassphraseDialog)
+
+	if dialog == nil {
+		return nil, fmt.Errorf("Failed to allocate a Confirmation of Installation Dialog")
+	}
+
+	if modelSI == nil {
+		return nil, fmt.Errorf("Missing model for Confirmation of Installation Dialog")
+	}
+
+	if err := initPassphraseDialogWindow(dialog); err != nil {
+		return nil, fmt.Errorf("Failed to create Confirmation of Installation Dialog: %v", err)
+	}
+
+	dialog.cancelButton.OnClick(func(ev clui.Event) {
+		dialog.Confirmed = false
+		dialog.Close()
+	})
+
+	dialog.confirmButton.OnClick(func(ev clui.Event) {
+		dialog.Confirmed = true
+		modelSI.CryptPass = dialog.passphraseEdit.Title()
```

Comment:
so we're dealing with plain passphrase?

Reply:
Yes, there is no hashed version of the passphrase. But, we DO NOT store the passphrase in the YAML file nor do we ever expose the plain text passphrase on a command line.

Reply:
ok